### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by MichaelEbert**
make parser callable. This reduces lambdas and increases rvalue-ability. Also added a debug build version.

### CODE REVIEW

**Comment by Idhrendur** [comment]
Could you add a test (or more) demonstrating how this would be used? I'm not clear on the usage.

**Comment by MichaelEbert** [comment]
its hard to demonstrate its usefulness in a test
but 
`registerKeyword("database", db_entry_parser_);`
where db_entry_parser is a parser, instead of having to do 
```
registerKeyword("database", [db_entry_parser_](std::istream& stream){db_entry_parser_.parse(stream));
```

File: tests/ParserTests.cpp [hunk 1]
```diff
@@ -50,6 +50,18 @@ TEST(Parser_Tests, KeywordsAreMatched)
...
+TEST(Parser_Tests, FunctionObjectEquivalentToParse)
+{
+	std::stringstream input{"key = value"};
+	std::string value;
+	commonItems::parser parser();
+	parser.registerKeyword("key", [this](std::istream& theStream) {
+		value = commonItems::getString(theStream);
+	});
+	parser(input);
+	ASSERT_EQ("value", value);
```
Comment (Idhrendur) [inline tests/ParserTests.cpp:62]:
```suggestion
	EXPECT_EQ("value", value);
```

**Review by Idhrendur — APPROVED** [review]

File: tests/ParserTests.cpp [hunk 2]
```diff
@@ -50,6 +50,18 @@ TEST(Parser_Tests, KeywordsAreMatched)
 	ASSERT_EQ("value", test.value);
 }
 
+TEST(Parser_Tests, FunctionObjectEquivalentToParse)
+{
+	std::stringstream input{"key = value"};
+	std::string value;
+	commonItems::parser parser{};
+	parser.registerKeyword("key", [this](std::istream& theStream) {
```
Comment (Idhrendur) [inline tests/ParserTests.cpp:58]:
```suggestion
	parser.registerKeyword("key", [&value](std::istream& theStream) {
```

**Review by Zemurin — APPROVED** [review]

**Pull request merged**